git_machete/github.py

```diff
@@ -90,71 +98,129 @@ def _token_from_hub() -> Optional[str]:
+            parsed_response_body: Any = json.loads(response.read().decode())
+            return parsed_response_body
+    except HTTPError as err:
+        if err.code == http.HTTPStatus.UNPROCESSABLE_ENTITY:
+            raise UnprocessableEntityHTTPError(err.reason)
         else:
-            first_line = fmt(f'GitHub API returned {response.status} HTTP status with error message: `{body.get("message")}`.\n')
+            first_line = fmt(f'GitHub API returned {err.code} HTTP status with error message: `{err.reason}`\n')
             if token:
-                raise MacheteException(
-                    first_line + fmt(f'Make sure that the token provided in `gh auth status` or `~/.config/hub` or <b>{GITHUB_TOKEN_ENV_VAR}</b> is valid '
-                                     f'and allows for access to `GET https://{host}{url}`.'))
+                raise MacheteException(first_line + fmt(f'Make sure that the token provided in `gh auth status` or `~/.config/hub` or <b>{GITHUB_TOKEN_ENV_VAR}</b> is valid and allows for access to `{method.upper()}` https://{host}{url}`.'))
             else:
                 raise MacheteException(
                     first_line + fmt(f'This repository might be private. Provide a GitHub API token with `repo` access via `gh` or `hub` or <b>{GITHUB_TOKEN_ENV_VAR}</b> env var.\n'
                                      'Visit `https://github.com/settings/tokens` to generate a new one.'))
     except OSError as e:
         raise MacheteException(f'Could not connect to {host}: {e}')
-    finally:
-        conn.close()
 
 
-def derive_pull_requests(org: str, repo: str) -> List[GitHubPullRequest]:
+def __check_pr_already_created(pull: GitHubPullRequest, pull_requests: List[GitHubPullRequest]) -> Optional[GitHubPullRequest]:
+    for pr in pull_requests:
+        if pull.base == pr.base and pull.head == pr.head:
+            return pr
+    return None
 
+
+def create_pull_request(org: str, repo: str, head: str, base: str, title: str, description: str, draft: bool) -> GitHubPullRequest:
     token: Optional[str] = github_token()
-    prs = fire_github_api_get_request(f'/repos/{org}/{repo}/pulls', token)
-    return [GitHubPullRequest(int(pr['number']), pr['user']['login'], pr['base']['ref'], pr['head']['ref']) for pr in prs]
+    request_body: Dict[str, Any] = {
+        'head': head,
+        'base': base,
+        'title': title,
+        'body': description,
+        'draft': draft
+    }
+    prs: List[GitHubPullRequest] = derive_pull_requests(org, repo)
+    to_load: GitHubPullRequest = GitHubPullRequest(1, 'user', base, head, '')
+    pr_found: Optional[GitHubPullRequest] = __check_pr_already_created(to_load, prs)
+    if not pr_found:
+        pr = __fire_github_api_request('POST', f'/repos/{org}/{repo}/pulls', token, request_body)
+        return parse_pr_json(pr)
+    else:
+        raise MacheteException(f'Pull request for branch {head} is already created under link {pr_found.html_url}! PR details:\n{pr_found}')
```

Comment:
Hmmm actually I looked at the output here:

![image](https://user-images.githubusercontent.com/3383210/129594284-70dc5328-c8a6-4155-8180-e6bfb359d5bc.png)

I'd change two things:
1. I don't think there's a need to print the status if no push was needed,
2. PR details are probably redundant here (or at least `\n` should go BEFORE `PR details`)